app/models/version.rb

```diff
@@ -26,4 +34,22 @@ def to_s
   def self.past_month_activities(objects, donor_id)
     related_to_multiple(objects).except_user(donor_id).past_month
   end
+
+  # required by PushUpdates and PaperTrail modules
+  def offer
+    related_type == "Offer" ? related : nil
+  end
+
+  def valid_pusher_update?
+    item_log? || item_package_log?
+  end
+
+  def item_log?
+    self.item_type == "Item" &&
+    (self.object_changes.keys & %w(donor_description donor_condition_id state)).present?
+  end
+
+  def item_package_log?
+    self.item_type == "Package" && %w(received missing).include?(self.object_changes["state"][1])
+  end
```

Comment:
@sunil-sharma I wonder if it is possible to use the following papertrail setup to exclude some events. https://github.com/airblade/paper_trail#choosing-attributes-to-monitor

E.g. if we only care about changes to donor_description, donor_condition_id and state then we could use :skip or :only

For item_package_log, I think it would be better to send all Version changes to Ember (via PushUpdates) and filter on the client side. This is because we will probably want to Versions for another feature one day and we will run into problems if we are only sending certain updates from the API.